tests/alt-registry.rs

```diff
@@ -262,3 +261,50 @@ fn alt_registry_and_crates_io_deps() {
 [FINISHED] dev [unoptimized + debuginfo] target(s) in [..] secs"))
 
 }
+
+#[test]
+fn block_publish_due_to_no_token() {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+        "#)
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    // Setup the registry by publishing a package
+    Package::new("bar", "0.0.1").alternative(true).publish();
+
+    // Now perform the actual publish
+    assert_that(p.cargo("publish").masquerade_as_nightly_cargo()
+                 .arg("--registry").arg("alternative").arg("-Zunstable-options"),
+                execs().with_status(101));
```

Comment:
Could you add an assertion about stderr here to ensure it prints the right error message?